### ISSUE TITLE
Backspace characters make pasting from a register fail. 

### ISSUE THREAD

**Issue by fiveNinePlusR** [emacs-evil/evil#1038]
# Issue type
- Bug report

# Environment

Emacs version: GNU Emacs 25.1.1 (x86_64-apple-darwin15.6.0, Carbon Version 157 AppKit 1404.47) of 2017-01-05
Operating System: macos high sierra
Evil version: evil-git-0fa3658cd
Evil installation type: spacemacs
Graphical/Terminal: gui
Tested in a `make emacs` session (see [CONTRIBUTING.md](https://github.com/emacs-evil/evil/blob/master/CONTRIBUTING.md#bug-reports)): No

# Reproduction steps
- Start Emacs
press `qritestt<backspace>q`

then press `"rpq`

# Expected behavior
should paste a string

# Actual behavior
it errors because there was a backspace character in the string. `evil-vector-to-string` and naive string creation are the culprit.

# Further notes
none

**Comment by fiveNinePlusR** [comment]
seems they print out the termcap representation of the keys if they are unprintable except for some cases. 

`<80>bs` is the code for backspace ran a few other tests and the function keys are `k1-k9`, ` k;`, `F1-F19`

a full list is here http://vimdoc.sourceforge.net/htmldoc/term.html#builtin-terms

I would say that's an implementation detail of vim that probably doesn't make sense to fully emulate. whatever `edit-last-kbd-macro` does to turn characters into printable representations probably makes sense here too.

**Comment by wasamasa** [comment]
Indeed. We can't even properly emulate it because Vim displays control characters (a newline/carriage return) here while we display two normal characters (`^` followed by `J` or `M`). It seems that the keyboard macro editor uses `edmacro-format-keys` for this purpose, we can use it in case something is a vector. For the case of something being a string, not sure, I'd keep the existing code for that.
